packages/core/tests/anonymousSignIn.test.ts

```diff
@@ -0,0 +1,142 @@
+import faker from '@faker-js/faker'
+import { interpret } from 'xstate'
+import { waitFor } from 'xstate/lib/waitFor'
+import { createAuthMachine } from '../src/machines'
+import { Typegen0 } from '../src/machines/index.typegen'
+import { BASE_URL } from './helpers/config'
+import {
+  anonymousNetworkErrorHandler,
+  correctAnonymousHandler,
+  deamonymisationSuccessfulHandler
+} from './helpers/handlers'
+import server from './helpers/server'
+import CustomClientStorage from './helpers/storage'
+import { GeneralAuthState } from './helpers/types'
+
+type AuthState = GeneralAuthState<Typegen0>
+
+const customStorage = new CustomClientStorage(new Map())
+
+const authMachine = createAuthMachine({
+  backendUrl: BASE_URL,
+  clientUrl: 'http://localhost:3000'
+})
+
+const authService = interpret(authMachine)
+
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
+afterAll(() => server.close())
+
+beforeEach(() => {
+  authService.start()
+})
+
+afterEach(() => {
+  authService.stop()
+  customStorage.clear()
+  server.resetHandlers()
+})
+
+test(`should fail if network is unavailable`, async () => {
+  server.use(anonymousNetworkErrorHandler)
+
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches('authentication.signedOut.failed')
+  )
+
+  expect(state.context.errors).toMatchInlineSnapshot(`
+    {
+      "authentication": {
+        "error": "OK",
+        "message": "Network Error",
+        "status": 200,
+      },
+    }
+  `)
+})
+
+test(`should fail if server returns an error`, async () => {
+  server.use(anonymousNetworkErrorHandler)
+
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches('authentication.signedOut.failed')
+  )
+
+  expect(state.context.errors).toMatchInlineSnapshot(`
+    {
+      "authentication": {
+        "error": "OK",
+        "message": "Network Error",
+        "status": 200,
+      },
+    }
+  `)
+})
+
+test('should deanonymise a user with email and password', async () => {
+  server.use(correctAnonymousHandler, deamonymisationSuccessfulHandler)
```

Comment:
you can include these handlers in `handlers/server.ts` as default handlers for successful scenarios